sw/device/silicon_creator/rom_ext/rom_ext.c

```diff
@@ -317,61 +413,148 @@ static uintptr_t owner_vma_get(const manifest_t *manifest, uintptr_t lma_addr) {
           (uintptr_t)_owner_virtual_start_address + CHIP_ROM_EXT_SIZE_MAX);
 }
 
-static size_t dice_certs_buffer_space_remaining(void) {
-  return FLASH_CTRL_PARAM_BYTES_PER_PAGE -
-         (size_t)(dice_cert_obj.obj_p - dice_certs_page);
+// Get the size of the remaining tail space that is not processed yet.
+OT_WARN_UNUSED_RESULT
+static size_t dice_chain_get_tail_size(void) {
```

Comment:
could you group all the `dice_chain_*` functions together, and clearly comment which ones will be moved to a separate lib?